src/core/stream.js

```diff
@@ -738,16 +738,17 @@ var PredictorStream = (function PredictorStreamClosure() {
     var inbits = 0, outbits = 0;
     var pos = bufferLength;
     var i;
+    var bitMask;
 
-    if (bits === 1) {
+    if (bits === 1 && colors === 1) {
+      bitMask = (1 << colors) - 1;
       for (i = 0; i < rowBytes; ++i) {
-        var c = rawBytes[i];
-        inbuf = (inbuf << 8) | c;
-        // bitwise addition is exclusive or
-        // first shift inbuf and then add
-        buffer[pos++] = (c ^ (inbuf >> colors)) & 0xFF;
-        // truncate inbuf (assumes colors < 16)
-        inbuf &= 0xFFFF;
+        var c = rawBytes[i] ^ inbuf;
+        c ^= c >> 1;
+        c ^= c >> 2;
+        c ^= c >> 4;
+        inbuf = (c & bitMask) << (8 - colors);
```

Comment:
No need to use bitMask for it, and we know that colors == 1. Revert bitMask changes above/below, you can just use `inbuf = (c & 1) << 7;` here.